.github/workflows/test-all-packages.yml

```diff
@@ -119,35 +119,58 @@ jobs:
       #- name: yarn test (everything)
       #  run: yarn ${{ steps.vars.outputs.test }}
       - name: yarn test (access-token)
+        if: success() || failure()
         run: cd packages/access-token && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (assert)
+        if: success() || failure()
         run: cd packages/assert && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (deployment)
+        if: success() || failure()
         run: cd packages/deployment && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (ERTP)
+        if: success() || failure()
         run: cd packages/ERTP && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (governance)
+        if: success() || failure()
         run: cd packages/governance && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (import-manager)
+        if: success() || failure()
         run: cd packages/import-manager && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (notifier)
+        if: success() || failure()
         run: cd packages/notifier && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (same-structure)
+        if: success() || failure()
         run: cd packages/same-structure && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (sharing-service)
+        if: success() || failure()
         run: cd packages/sharing-service && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (sparse-ints)
+        if: success() || failure()
         run: cd packages/sparse-ints && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (spawner)
+        if: success() || failure()
         run: cd packages/spawner && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (stat-logger)
+        if: success() || failure()
         run: cd packages/stat-logger && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (store)
+        if: success() || failure()
         run: cd packages/store && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (swing-store)
+        if: success() || failure()
         run: cd packages/swing-store && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (cosmic-proto)
+        if: success() || failure()
         run: cd packages/cosmic-proto && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
+      - name: notify on failure
+        if: failure() && github.event_name != 'pull_request'
+        uses: ./.github/actions/notify-status
+        with:
+          webhook: ${{ secrets.SLACK_WEBHOOK_URL }}
```

Comment:
Given the number of times this `with:` block appears (20, by my count), would it make sense to replace `notify-status` with `notify-slack-and-email`, and to move these `with: secrets.*` lookups into the action? It would make the action less generic, but it's already living in our own repo anyways, and it's not like we're going to send email notifications to different addresses on a per-job basis. Or maybe have the action use e.g. `secrets.SLACK_WEBHOOK_URL` as the default, so each job could override it with like a specific slack channel, but you could otherwise cut out the 100 lines needed to provide the same secrets to every invocation.

Reply:
`secrets.*` are not available in actions.  You need to pass them explicitly.